core-api/src/main/java/com/optimizely/ab/Optimizely.java

```diff
@@ -495,20 +499,21 @@ public Double getFeatureVariableDouble(@Nonnull String featureKey,
             return null;
         }
 
-        String variableValue = getFeatureVariableValueForType(
-            featureKey,
-            variableKey,
-            userId,
-            attributes,
-            FeatureVariable.VariableType.DOUBLE
-        );
-        if (variableValue != null) {
-            try {
-                return Double.parseDouble(variableValue);
-            } catch (NumberFormatException exception) {
-                logger.error("NumberFormatException while trying to parse \"" + variableValue +
-                    "\" as Double. " + exception);
+        Double variableValue = null;
+        try {
+            variableValue = getFeatureVariableValueForType(
+                featureKey,
+                variableKey,
+                userId,
+                attributes,
+                FeatureVariable.VariableType.DOUBLE
+            );
+            if (variableValue != null) {
```

Comment:
This check seems redundant.  You can just change last line to be `return variableValue`